src/serializer.js

```diff
@@ -14,12 +14,23 @@ function createSerializer(styleSheet) {
     val.withStyles = true
     const printedVal = printer(val)
     if (styles) {
-      return `${styles}\n\n${printedVal}`
+      return replaceClassNames(selectors, styles, printedVal)
     } else {
       return printedVal
     }
   }
 
+  function replaceClassNames(selectors, styles, code) {
+    let index = 0
+    return selectors.reduce((acc, className) => {
+      const escapedRegex = new RegExp(
```

Comment:
Why don't we put this logic in its own module and test it separately. That way we can get high confidence it works for the cases we need without writing a lot of confusing tests for the serializer.